### DISCUSSION TITLE
Registration

### DISCUSSION THREAD

**Discussion by patrik-cihal** [maxcountryman/axum-login#188] [Q&A]
Sorry, I'm quite new both to Axum and this crate, but how do I even in the simple sqlite example, create a post request with the option to register? I understand that probably inserting it into the DB is enough, but then how do I get access to the DB from `async fn register()` function?

**Comment by patrik-cihal** [comment]
Solved! Just edit the `Backend` struct in `users.rs`, there you have access to `db`.

**Reply by maxcountryman** [reply]
That's a perfectly reasonable solution, absolutely. 💯

**Comment by maxcountryman** [comment]
We've omitted user registration to brevity but perhaps it would be helpful to include that as an illustration.

While there's various patterns that will help with this, typically what I've seen is people will put a database connection pool into [an application state that's passed into the handler](https://docs.rs/axum/latest/axum/#sharing-state-with-handlers). With this, you could accept a structure form of registration details and insert a new user row into the database.
